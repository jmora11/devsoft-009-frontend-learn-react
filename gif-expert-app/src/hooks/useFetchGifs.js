import { useEffect, useState } from "react"
import { gifFetch } from "../helpers/getGifs";

export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        gifFetch(category)
            .then( imgs => {

                setTimeout(()=> {
                    setState({
                        data: imgs,
                        loading: false
                    })
                }, 2000)

            });
    }, [ category ])

    return state;
}