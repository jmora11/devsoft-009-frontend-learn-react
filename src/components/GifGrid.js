import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data: imagenes, loading } = useFetchGifs(category);

    return (
        <>
            <h1 className='animate__animated animate__bounce'>{category}</h1>

            { loading && <p>Loading...</p> }

            <div className='card-grid'>
                {
                    imagenes.map( img =>
                        <GifGridItem
                            key={ img.id }    
                            { ...img }
                        />  
                    )
                }
            </div>
        </>
    )
};
