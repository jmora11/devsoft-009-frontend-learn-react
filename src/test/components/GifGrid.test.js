import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Gif Grid', () => {

    test('debe mostrar correctamente el snapshot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid category={ 'Demon Slayer' } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar items el snapshot', () => {

        const imgs = [{
            id: 'ABC',
            url: 'http://mock:img',
            title: 'Titke Mock'
        }];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={ 'Demon Slayer' } /> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('no debe mostrar parrafo loading', () => {

        const imgs = [
            {
                id: 'ABC',
                url: 'http://mock:img1',
                title: 'Titke Mock 1'
            },
            {
                id: 'DEF',
                url: 'http://mock:img2',
                title: 'Title Mock 2'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: imgs,
            loading: false
        });
        const wrapper = shallow( <GifGrid category={ 'Demon Slayer' } /> );
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( imgs.length );
    });
});