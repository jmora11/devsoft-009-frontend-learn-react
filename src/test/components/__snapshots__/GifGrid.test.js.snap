// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Grid debe mostrar correctamente el snapshot 1`] = `
<Fragment>
  <h1
    className="animate__animated animate__bounce"
  >
    Demon Slayer
  </h1>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Gif Grid debe mostrar items el snapshot 1`] = `
<Fragment>
  <h1
    className="animate__animated animate__bounce"
  >
    Demon Slayer
  </h1>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Titke Mock"
      url="http://mock:img"
    />
  </div>
</Fragment>
`;
